### PR TITLE
fix: name conflict and `GetGlobalOperator`

### PR DESCRIPTION
**Pull request by Vipul-Cariappa**
Fixes 5 previously failing test cases.

### CODE REVIEW

File: clingwrapper/src/clingwrapper.cxx [hunk 1]
```diff
@@ -477,7 +477,7 @@ bool Cppyy::AppendTypesSlow(const std::string &name,
   if (!struct_count)
     Cpp::Declare(code.c_str()); // initialize the trampoline
 
-  std::string var = "__s" + std::to_string(struct_count++);
```
Comment (Vipul-Cariappa) [inline clingwrapper/src/clingwrapper.cxx:480]: `__s8` conflicts with some variable/type name.

File: clingwrapper/src/cpp_cppyy.h [hunk 2]
```diff
@@ -311,7 +311,7 @@ namespace Cppyy {
 
     RPY_EXPORTED
     TCppIndex_t  GetGlobalOperator(
-        TCppType_t scope, const std::string& lc, const std::string& rc, const std::string& op) { assert(0 && "GetGlobalOperator");return 0; }
+        TCppType_t scope, const std::string& lc, const std::string& rc, const std::string& op) { assert(0 && "GetGlobalOperator");return -1; }
```
Comment (Vipul-Cariappa) [inline clingwrapper/src/cpp_cppyy.h:314]: `GetGlobalOperator` is called by `BuildOperator` at https://github.com/Vipul-Cariappa/CPyCppyy-compiler-research/blob/b59c6b1846b40ba99fc2c5d82cbbfd90401d1667/src/Utility.cxx#L271-L274.

I believe it should be -1 and not 0. By looking at the logic at caller end.

**Comment by Vipul-Cariappa** [comment]
ping @vgvassilev @aaronj0

**Review by aaronj0 — APPROVED** [review]

**Comment by aaronj0** [comment]
Looks good to me! Let's wait for the checks to finish before merging

**Comment by Vipul-Cariappa** [comment]
It is failing with OSX cling.
I do not have access you any device with OSX. I cannot test it.
Can we set `run=not IS_MAC` for now?
I guess `test_stltypes.py::TestSTLSTRING::test04_array_of_strings` is failing from the logs.

**Comment by aaronj0** [comment]
> It is failing with OSX cling. I do not have access you any device with OSX. I cannot test it. Can we set `run=not IS_MAC` for now? I guess `test_stltypes.py::TestSTLSTRING::test04_array_of_strings` is failing from the logs.

Was this a previously passing test

> It is failing with OSX cling. I do not have access you any device with OSX. I cannot test it. Can we set `run=not IS_MAC` for now? I guess `test_stltypes.py::TestSTLSTRING::test04_array_of_strings` is failing from the logs.

Looks like it crashes now, instead of just failing previously. I would edit the tag to use `run = not(IS_MAC and not IS_CLANG_REPL)` to no longer run on OS X cling

**Pull request merged**